### PR TITLE
Add config

### PR DESCRIPTION
**Pull request by bmath10**
**Notes**:
1. Run `yarn install` because new packages have been added.
2. I followed [this tutorial](https://codingsans.com/blog/configure-frontend-projects-with-dotenv) for reading in environment variables in an Angular project after doing research and trying a few other things.

--------------

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-28
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:** @mayerm94

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [x] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge is complete.

### CODE REVIEW

File: src/app/components/home-page/home-page.component.spec.ts [hunk 1]
```diff
@@ -37,31 +37,31 @@ describe('HomePageComponent', () => {
...
     it('should render unresponsive iframe fourth', () => {
-      expect(iframeElements[3].src).toEqual(component.unresponsive);
+      expect(iframeElements[3].src).toEqual(component.getUrl(component.unresponsive));
     });
 
-    it('should render new endpoints iframe fifth', () => {
-      expect(iframeElements[4].src).toEqual(component.newEndpoints);
-    });
+    // it('should render new endpoints iframe fifth', () => {
+    //   expect(iframeElements[4].src).toEqual(component.getUrl(component.newEndpoints));
```
Comment (mayerm94) [inline src/app/components/home-page/home-page.component.spec.ts]: Why was this test commented out?

Comment (mayerm94) [reply src/app/components/home-page/home-page.component.spec.ts]: nvm I think I see why, because  `// public newEndpoints = '&panelId=20';`

Comment (bmath10) [reply src/app/components/home-page/home-page.component.spec.ts]: yeah i just commented out the new endpoints vis since it's currently just a placeholder and throws a bunch of errors since it doesn't actually work.

Comment (mayerm94) [reply src/app/components/home-page/home-page.component.spec.ts]: Followup question, if the `newEndpoints` viz is not in the works can we just delete the panelID definition, html entry and the test? Or atleast TODO them?

Comment (bmath10) [reply src/app/components/home-page/home-page.component.spec.ts]: sure, i can do that!

File: README.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # lantern-front-end
 
+## Configuration
+This project reads the following environment variable:
+
+* **GRAFANA_URL**: The grafana dashboard url that is being used for displaying visualizations.
```
Comment (edeyoung) [inline README.md]: i think this should include an application-based lead-in, like, "LANTERN_GRAFANA_URL". Obviously, "lantern" is subject to change, but you could call it that for now so when we do the find/replace, it will be caught.

**Review by edeyoung — COMMENTED** [review]
I had one comment. I don't have a good handle on best practices for angular, so didn't feel able to comment in a few places. Looks good to me from what I understood.

File: README.md (hunk 2 above)
Comment (bmath10) [reply README.md]: Makes sense to me!

File: config/config-loader.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ const config = convict({
   grafana_url: {
     default: "",
```
Comment (edeyoung) [inline config/config-loader.js]: there's no default value although the README says there is

File: config/config-loader.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+const convict = require("convict");
+const dotenv = require("dotenv");
+
+// Reads in the environment variables and saves it into the process.env global node variable
+dotenv.config();
+
+// Create a configuration schema
+const config = convict({
+  grafana_url: {
+    default: "",
```
Comment (edeyoung) [inline config/config-loader.js:12]: there's no default although the README says htere is

**Comment by edeyoung** [comment]
I got the following output when running the e2e test:
image

File: config/config-loader.js (hunk 4 above)
Comment (bmath10) [reply config/config-loader.js:12]: Oh, yeah you're totally right, I'll update the README

**Comment by bmath10** [comment]
> I got the following output when running the e2e test:
> image

I reran the tests and didn't get any errors, even after removing my node_modules and reinstalling. Could you try increasing the number on line 23 in ```e2e/protractor.conf.js``` just to see if that changes anything? If it doesn't, then it's probably not just a timeout issue, and I'll keep looking to see if there's something else going on. If @mayerm94 wants to try it too that would be helpful!

**Review by mayerm94 — APPROVED** [review]
I could not replicate the failing tests issue that @edeyoung  was experiencing

**Comment by edeyoung** [comment]
> > I got the following output when running the e2e test:
> > image
> 
> I reran the tests and didn't get any errors, even after removing my node_modules and reinstalling. Could you try increasing the number on line 23 in `e2e/protractor.conf.js` just to see if that changes anything? If it doesn't, then it's probably not just a timeout issue, and I'll keep looking to see if there's something else going on. If @mayerm94 wants to try it too that would be helpful!

figured it out - test worked. approving pr.

**Review by edeyoung — APPROVED** [review]
